Kudu.Core/Jobs/JobsManagerBase.cs

```diff
@@ -495,6 +496,16 @@ private static string FindCommandToRun(FileInfoBase[] files, out IScriptHost scr
                         }
                     }
                 }
+
+                foreach (var supportedFileName in scriptHost.SupportedFileNames)
```

Comment:
I don't see any unit tests for any of this tricky logic. That would be very useful - tests that verify this.